server/src/controllers/Request.js

```diff
@@ -199,3 +199,62 @@ export const getRemainingRequestsByCompany = async (req, res) => {
     });
   }
 };
+
+export const updateFile = async (req, res) => {
+  try {
+    const requestId = req.get("requestId");
+    const fileName = req.get("fileName");
+    const userId = req.get("userId");
+    const filePath = `${userId}/request/${requestId}/${fileName}`;
+
+    // console.log(filePath);
+
+    const request = await Request.findByIdAndUpdate(
+      requestId,
+      { $push: { request_files: filePath } },
+      { returnOriginal: false },
+    );
+
+    // console.log(request);
+    res.status(200).json({
+      message: "Upload file successfully",
+      data: request,
+    });
+  } catch (error) {
+    res.status(400).json({
+      message: "Error storing file path",
+      error: error.message,
+    });
+  }
+};
+
+export const deleteFile = async (req, res) => {
+  try {
+    const { path } = req.body;
+
+    const request_id = path.split("/")[2];
+    const request = await Request.findById(request_id);
+
+    const new_request_files = request.request_files.filter((e) => e != path);
```

Comment:
The solution works here, but I think it can be a bit [cleaner](https://stackoverflow.com/a/27917378). Additional upside is that there's 1 less network call and the update is atomic.